### ISSUE TITLE
[V5] Theme override object typing issue

### ISSUE THREAD

**Issue by 7iomka** [mantinedev/mantine#1723]
### What package has an issue

@mantine/core

### Describe the bug

When I store theme override in an object like [docs says](https://v5.mantine.dev/theming/theme-object/#store-theme-override-object-in-a-variable) I received error in some cases, for example in case of `params`

```ts
 Type 'unknown' is not assignable to type 'ButtonStylesParams'
```

Repro: https://codesandbox.io/s/nice-poitras-i81z4v?file=/src/App.tsx

### In which browser did the problem occur

Chrome

**Comment by rtivital** [comment]
Thanks for reporting

**Comment by rtivital** [comment]
Fixed in 5.0
